### PR TITLE
Update infrastructure monitoring docs about containers

### LINKED ISSUE #3745: [Request]: Update Containers docs for Infrastructure
### Description

We're going to be updating our container template to (a) use the new template we are using for hosts (b) making 'docker containers' support **all** containers.

Initially, this means we need to:

1. Update the [metrics definitions](https://www.elastic.co/guide/en/observability/current/docker-container-metrics.html)

2. Update the [inventory docs](https://www.elastic.co/guide/en/observability/current/view-infrastructure-metrics.html) to reflect these are 'containers' (not just 'docker containers')

_Old view of containers_
containers - now

_What it will start to look like (showing 'hosts' for now)_
containers - next

**Notes:** 
- See [Docker Container Metric Definitions](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.lrmgncaurmh9) and [Kubernetes Container Metric Definitions](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.k69qexcnfbv8)
- There will be future releases which start to incorporate 'containers' into other views (e.g. from a host or a service) but that's out of scope for now (see [epic for details](https://github.com/elastic/observability-dev/issues/3470))

### Resources

- [x] https://github.com/elastic/observability-dev/issues/3313

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

None

### What release is this request related to?

TBC  (likely 8.15)

### Collaboration model

I'm happy to help!

### Point of contact.

**Main contact:** @roshan-elastic  + @jennypavlova + @MiriamAparicio 

**Stakeholders:**
@smith

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #3745 

Preview links:
* https://observability-docs_bk_4186.docs-preview.app.elstc.co/guide/en/observability/master/view-infrastructure-metrics.html
* https://observability-docs_bk_4186.docs-preview.app.elstc.co/guide/en/observability/master/container-metrics.html

TODO after merging:

- [ ] Port changes to serverless
- [ ] Set up redirect for changed path

### CODE REVIEW

**Review by jennypavlova — APPROVED** [review]
Nice 👏  thank you! 
One note regarding the metrics - we also show `Network` and `Disk` now: 

Screenshot 2024-08-22 at 19 05 41
Screenshot 2024-08-22 at 19 05 25

(Not sure if we want to include it in the screenshots - cc @roshan-elastic)
Overall LGTM but I am leaving the final review to @roshan-elastic

**Comment by dedemorton** [comment]
>One note regarding the metrics - we also show Network and Disk now

@jennypavlova Good point! I didn't see those tabs because I was looking at an environment that was all K8s. I think I should show those tabs.

**Comment by dedemorton** [comment]
@jennypavlova or @roshan-elastic I have a few follow-up questions after applying the Jenny's changes:

- Can you point me to an environment that shows docker metrics? I've only been able to find oblt environments with K8s. I've temporarily used the screenshot that Jenny created to show all the metrics, but I think I might also need to redo the overview screenshot.
- Are network and disk metrics shown in 8.15 as well as 8.16?
- I don't see disk metrics listed for Docker in this document. Should it be? If so, I think we need to update the topic about docker metrics, too.

**Comment by roshan-elastic** [comment]
> Can you point me to an environment that shows docker metrics? I've only been able to find oblt environments with K8s. I've temporarily used the screenshot that Jenny created to show all the metrics, but I think I might also need to redo the overview screenshot.

Think I found something on [edge](https://edge-oblt.kb.us-west2.gcp.elastic-cloud.com/app/metrics/inventory?inventoryViewId=%270%27&waffleFilter=(expression:%27container.runtime%20:%20%22docker%22%27,kind:kuery)&waffleTime=(currentTime:1724746043969,isAutoReloading:!f)&waffleOptions=(accountId:%27%27,autoBounds:!t,boundsOverride:(max:1,min:0),customMetrics:!(),customOptions:!((field:container.runtime,text:container.runtime)),groupBy:!((field:container.runtime)),legend:(palette:cool,reverseColors:!f,steps:10),metric:(type:cpu),nodeType:container,region:%27%27,sort:(by:name,direction:desc),timelineOpen:!f,view:map)&assetDetailsFlyout=(assetType:!n,detailsItemId:!n)&assetDetails=!n):

image


> Are network and disk metrics shown in 8.15 as well as 8.16?

Checking on the demo cluster (8.15), it doesn't look like it. Looks like 8.16 only.

> I don't see disk metrics listed for Docker in [this document](https://docs.google.com/document/d/1xrLMnT-Nz5TqkPT5PxNjJNLjrU4qXMsbyTWkmqLhtLw/edit#heading=h.lrmgncaurmh9). Should it be? If so, I think we need to update the topic about docker metrics, too.

Good catch. Added it now.

One note - it looks like the `docker` metrics are being used even for `containerd` containers - this **looks** like a bug so I'd ignore documenting Kubernetes network/disk metrics for now.

@jennypavlova - Could you confirm if this is a bug? I know we're going to clean all of this up in https://github.com/elastic/observability-dev/issues/3470

**Comment by dedemorton** [comment]
This PR targets 8.15, so I am going to remove mention of network and disk metrics for now.

I'll open a separate issue for 8.16 to make sure network and disk metrics get added when we're ready to document them.

File: docs/en/observability/monitor-infra/container-metrics.asciidoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+| **Inbound Traffic (RX)**            | Derivative of the maximum of `docker.network.in.bytes` scaled to a 1 second rate.
+
+| **Outbound Traffic (TX)**           | Derivative of the maximum of `docker.network.out.bytes` scaled to a 1 second rate.
+|===
+
+[discrete]
+[[key-metrics-kubernetes]]
+== Kubernetes container metrics
+
+These are the key metrics displayed for Kubernetes (ContainerD) containers.
```
Comment (colleenmcginnis) [inline docs/en/observability/monitor-infra/container-metrics.asciidoc:58]: If you end up changing the other mention, you should change this one, too. 

```suggestion
These are the key metrics displayed for Kubernetes (containerd) containers.
```

**Review by colleenmcginnis — COMMENTED** [review]
Just a tiny suggestion.

**Review by colleenmcginnis — APPROVED** [review]
🐥

**Pull request merged**